bigchaindb/commands/bigchaindb.py

```diff
@@ -162,13 +150,26 @@ def run_drop(args):
         print("Cannot drop '{name}'. The database does not exist.".format(name=dbname), file=sys.stderr)
 
 
+def run_recover(b):
+    pre_commit = query.get_pre_commit_state(b.connection, PRE_COMMIT_ID)
+
+    # Initially the pre-commit collection would be empty
+    if pre_commit:
+        latest_block = query.get_latest_block(b.connection)
```

Comment:
We store the `pre_commit` **before** storing the `block`.

In the unfortunate event of BigchainDB crashing before writing the very first block, `latest_block` would be `null`, and accessing `latest_block['height']` would generate a `KeyError`. Is this a possibility?

Reply:
That was fast, thanks!